### ISSUE TITLE
Abtract class with generic type x extends y results in invalid output

### ISSUE THREAD

**Issue by kiliancs** [alangpierce/sucrase#397]
Using sucrase 3.9.3, an abstract class with a generic type extending another type results in an output that calls super. This is invalid since it's not a subclass.

```ts
abstract class AbstractClassWithGenericExtends<T extends Y> {
  public x = 1;
}
```

Result:

```js
class AbstractClassWithGenericExtends {constructor(...args) { super(...args); AbstractClassWithGenericExtends.prototype.__init.call(this); }
   __init() {this.x = 1}
}
```

https://sucrase.io/#code=abstract%20class%20AbstractClassWithGenericExtends%3CT%20extends%20Y%3E%20%7B%0A%20%20public%20x%20%3D%201%3B%0A%7D

**Comment by kiliancs** [comment]
I think this one is important because all the other issues I found have a workaround (for example, use function instead of arrow function), but this one doesn't have an obvious workaround.

**Comment by alangpierce** [comment]
Makes sense, thanks for all of the bug reports! I *think* this one shouldn't be that bad, so I'll see if I can take a crack at it.

**Comment by alangpierce** [comment]
Just released a fix in 3.9.4.
